web/src/features/VersionDiff/ViewDiffButton.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { Version } from "@types";
+import { useSelectedApp } from "@features/App/hooks/useSelectedApp";
+import { useIsHelmManaged } from "@components/hooks";
+
+interface Props {
+  onWhyNoGeneratedDiffClicked: (version: Version) => void;
+  onWhyUnableToGeneratedDiffClicked: (version: Version) => void;
+  onViewDiffClicked: (firstSequence: number, secondSequence: number) => void;
+  version: Version;
+  versionHistory: Version[];
+}
+
+interface DiffSummary {
+  filesChanged?: number;
+}
+
+// TODO: unmarshal this in the fetch handler
+function unmarshallDiffSummary(diffSummary: string): DiffSummary {
+  try {
+    return JSON.parse(diffSummary) || {};
+  } catch (err) {
+    throw err;
+  }
+}
```

Comment:
there's quite a bit of the json in json fun times in these apis as well. I will unmarshall them in place in the fetch handler when i build the hook. no reason to spread `JSON.parse` around 

https://github.com/replicatedhq/kots/blob/d8ab2b5cc500cd296ab19fb82dc5f7ef05404196/web/src/components/apps/AppVersionHistory.tsx#L495-L504

also simplified it a bit into `numberOfFilesChanged` since that's actually the concept we're dealing with here. I will eventually move this into a selector and change it to something like `showViewDiffButton` instead of using this `>` expression 

https://github.com/replicatedhq/kots/blob/6cf7cc3fde758042fdca3092f3158cbcd2ff2f38/web/src/features/VersionDiff/ViewDiffButton.tsx#L51-L53